librz/bin/bobj.c

```diff
@@ -242,29 +243,32 @@ static RzList *classes_from_symbols(RzBinFile *bf) {
 }
 
 // TODO: kill offset and sz, because those should be inferred from binfile->buf
-RZ_IPI RzBinObject *rz_bin_object_new(RzBinFile *bf, RzBinPlugin *plugin, ut64 baseaddr, ut64 loadaddr, ut64 offset, ut64 sz) {
+RZ_IPI RzBinObject *rz_bin_object_new(RzBinFile *bf, RzBinPlugin *plugin, RzBinObjectLoadOptions *opts, ut64 offset, ut64 sz) {
 	rz_return_val_if_fail(bf && plugin, NULL);
 	ut64 bytes_sz = rz_buf_size(bf->buf);
 	Sdb *sdb = bf->sdb;
 	RzBinObject *o = RZ_NEW0(RzBinObject);
 	if (!o) {
 		return NULL;
 	}
+	o->opts = *opts;
+	if (o->opts.loadaddr == UT64_MAX) {
+		// no loadaddr means 0 loadaddr
```

Comment:
Please note, that in `rz_core_fileopen()` and `rz_core_bin_load()` the `loadaddr == UT64_MAX` means autodetection, not the `0` loadaddr. Might be useful to point the distinction.

Reply:
Thanks, that is a good pointer. I will re-check what the loadaddr means in detail here.

Reply:
I can't see any autodetection of the laddr in these functions, from what I can tell it just gets passed into `rz_io_map_new()` (Except in `rz_core_file_open_many()` where it is used as the baddr somehow but that looks very wrong). Do you maybe mean the baddr autodetection?

Reply:
I was referring exactly to this:
> Except in rz_core_file_open_many() where it is used as the baddr somehow but that looks very wrong.

Also this: https://github.com/rizinorg/rizin/blob/dev/librz/core/cfile.c#L1202:
```c
/* loadaddr is r2 -m (mapaddr) */
RZ_API RzCoreFile *rz_core_file_open(RzCore *r, const char *file, int flags, ut64 loadaddr) {
...
	//used by rz_core_bin_load otherwise won't load correctly
	//this should be argument of rz_core_bin_load <shrug>
	if (loadaddr != UT64_MAX) {
		rz_config_set_i(r->config, "bin.laddr", loadaddr);
	}
	rz_core_cmd0(r, "R!");
beach:
	r->times->file_open_time = rz_time_now_mono() - prev;
	return fh;
}
```

Reply:
Ok, then as far as I can tell the comment in that specific place is correct. Still a lot to improve for laddr stuff probably.